Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -968,4 +968,86 @@ u8 iso_day(Object& temporal_object)
     VERIFY_NOT_REACHED();
 }
 
+// 12.1.45 DefaultMergeFields ( fields, additionalFields ), https://tc39.es/proposal-temporal/#sec-temporal-defaultmergefields
+Object* default_merge_fields(GlobalObject& global_object, Object& fields, Object& additional_fields)
+{
+    auto& vm = global_object.vm();
+
+    // 1. Let merged be ! OrdinaryObjectCreate(%Object.prototype%).
+    auto* merged = Object::create(global_object, global_object.object_prototype());
+
+    // 2. Let originalKeys be ? EnumerableOwnPropertyNames(fields, key).
+    auto original_keys = fields.enumerable_own_property_names(Object::PropertyKind::Key);
+    if (vm.exception())
+        return {};
+
+    // 3. For each element nextKey of originalKeys, do
+    for (auto& next_key : original_keys) {
+        // a. If nextKey is not "month" or "monthCode", then
+        if (next_key.as_string().string() != vm.names.month.as_string() && next_key.as_string().string() != vm.names.monthCode.as_string()) {
+            auto property_name = PropertyName::from_value(global_object, next_key);
+
+            // i. Let propValue be ? Get(fields, nextKey).
+            auto prop_value = fields.get(property_name);
+            if (vm.exception())
+                return {};
+
+            // ii. If propValue is not undefined, then
+            if (!prop_value.is_undefined()) {
+                // 1. Perform ! CreateDataPropertyOrThrow(merged, nextKey, propValue).
+                merged->create_data_property_or_throw(property_name, prop_value);
+            }
+        }
+    }
+
+    // 4. Let newKeys be ? EnumerableOwnPropertyNames(additionalFields, key).
+    auto new_keys = additional_fields.enumerable_own_property_names(Object::PropertyKind::Key);
+    if (vm.exception())
+        return {};
+
+    // 5. For each element nextKey of newKeys, do
+    for (auto& next_key : new_keys) {
+        auto property_name = PropertyName::from_value(global_object, next_key);
+
+        // a. Let propValue be ? Get(additionalFields, nextKey).
+        auto prop_value = additional_fields.get(property_name);
+        if (vm.exception())
+            return {};
+
+        // b. If propValue is not undefined, then
+        if (!prop_value.is_undefined()) {
+            // i. Perform ! CreateDataPropertyOrThrow(merged, nextKey, propValue).
+            merged->create_data_property_or_throw(property_name, prop_value);
+        }
+    }
+
+    // 6. If newKeys does not contain either "month" or "monthCode", then
```

Comment:
Instead of iterating over the new_keys list 3 times (aka the loop above and these two calls) we can just add a
```c++
bool has_seen_month = false;
```
and update it in the for loop, aka
```c++
has_seen_month = has_seen_month || next_key.as_string().string() == vm.names.month.as_string() || next_key.as_string().string() == vm.names.monthCode.as_string();
```
but this is totally optional since this is just an optimisation :)

Reply:
Great idea, done! (using a slightly more verbose name :D)